README.md

```diff
@@ -29,10 +29,203 @@ Once you have your Swift package set up, adding ForAPIKit as a dependency is as
+        return nil
+    }
+    
+    var queryParameters: [String: Any]? {
+        // Define query parameters for each API endpoint if needed
+        return nil
+    }
+    
+    var keypathToMap: String? { 
+        // Define query keypathToMap for each API endpoint if needed
+        return nil 
+    }
+}
+```
+
+#### <i class="icon-file"></i>**JSONParsing**
+The `JSONParseable` or `Codable` protocol provided by FoxAPIKit allows you to easily create custom model objects in Swift from `JSON` data received from an API response. It simplifies the process of parsing and mapping JSON data to your model objects, making it efficient and convenient.
+```swift
+import JSONParsing
+
+// Confirm `JSONParseable` protocol
+struct User: JSONParseable {
+    
+    var id: Int
+    var name: String
+    var weight: Double?
+    var accounts: [String]
+    
+    static func parse(_ json: JSON) throws -> User {
+        return try User(
+            id: json["id"]^,
+            name: json["name"]^!,
+            weight: json["weight"]^?,
+            accounts: json["accounts"]^^
+        )
+    }
+}
+
+// Confirm `Codable` protocol
+struct User: Codable {
+    
+    var id: Int
+    var name: String
+    var weight: Double?
+    var accounts: [String]
+    
+    enum CodingKeys: CodingKey {
+        case id
+        case name
+        case weight
+        case accounts
+    }
+}
+```
+#### <i class="icon-file"></i>**Error Handling**
+Error Handling
+The APIClient provides built-in error handling for common HTTP errors, such as invalid URLs, network errors, and server errors. Additionally, you can define and handle custom API errors using the APIError enum provided by FoxAPIKit. Here's an example:
+
+```swift
+import FoxAPIKit
+
+enum MyAPIError: Error {
+    case invalidResponse
+    case unauthorized
+    case customError(message: String)
+}
+```
+#### <i class="icon-file"></i>**API Client**
+The APIClient is a powerful and customizable HTTP client provided by FoxAPIKit that handles making API requests and handling responses in your Swift-based iOS applications. It provides a simple and convenient way to communicate with APIs and retrieve data from remote servers.
+
+```swift
+class MyAPIClient: APIClient<AuthHeaders, ErrorResponse> {
+    
+    static let shared = SSAPIClient()
+    
+    override init() {
+        super.init()
+        #if DEBUG
+        enableLogs = true
+        #else
+        enableLogs = false
+        #endif
+    }
+    
+    override func authenticationHeaders(response: HTTPURLResponse) -> AuthHeaders? {
+        return self.authHeaders
+    }
+}
+```
+#### <i class="icon-file"></i>**Authentication**
+If your API requires authentication, you can customize the headers of the API request by setting the headers property of the `APIClient` instance. For example:
+
+```swift
+// Set custom headers for authentication
+SSAPIClient.shared.authHeaders = AuthHeaders(token: <your_auth_token>)
```

Comment:
@vinayak-nickelfox Since this is generic, can you replace it with 'FLAPIClient' or simple `APIClient`

Reply:
@vaibhav-nickelfox Since I'm using `MyAPIClient` in example code snippet i named it that way.